openapi.yaml

```
@@ -1926,6 +1926,283 @@ paths:
+      requestBody:
+        required: true
+        content:
+          multipart/form-data:
+            schema:
+              type: object
+              properties:
+                alias:
+                  type: string
+                  description: Name of the secret.
+                description:
+                  type: string
+                  description: Description of the secret.
+                value:
+                  type: string
+                  description: The content of the secret.
+                allowed_workspaces_type:
+                  type: string
+                  description: Which workspaces in this organization are allowed to use this secret in custom functions.
+                  enum:
+                    - ALL
+                    - SOME
+                    - NONE
+                allowed_workspaces:
+                  type: array
+                  description: Which workspaces are allowed if only SOME workspaces are allowed.
+                  items:
+                    type: string
+              required:
+                - alias
+                - value
+                - allowed_workspaces_type
+      responses:
+        '200':
+          description: Secret created successfully.
+        default:
+          description: Error response.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+      x-fern-examples:
+        - code-samples:
+            - sdk: curl
+              code: |
+                  curl "${API_ROOT}/v2/secrets" \
+                    -H "Authorization: Bearer ${API_TOKEN}" \
+                    -H "IB-Context: ${IB_CONTEXT}"\
+                    -F alias="my_secret" \
+                    -F description="Sample secret" \
+                    -F content="my password" \
+                    -F allowed_workspaces_type="SOME" \
+                    -F allowed_workspaces="workspace1" \ # Only required when allowed_workspaces_type="SOME"
+                    -F allowed_workspaces="workspace2"
+            - sdk: python
+              code: |
+                  # AI Hub Python SDK example
+                  from aihub import AIHub
+
+                  client = AIHub(api_root="<API-ROOT>",
+                                  api_key="<API-TOKEN>",
+                                  ib_context="<IB-CONTEXT>")
+
+                  result = client.secrets.create(
+                    alias="my_secret",
+                    description="Sample secret",
+                    content="my password"
+                    allowed_workspaces_type="SOME",
+                    allowed_workspaces=['workspace1', 'workspace2']) # Only required when allowed_workspaces_type="SOME"
+
+    patch:
+      operationId: updateSecret
+      x-fern-audiences:
+        - public
+      tags:
+        - secrets
+      summary: Update secret
+      description: |
+        Update the value, description, or allowed workspaces for a secret.
```

```suggestion
        Update the description, value, or allowed workspaces for a secret.
```

To preserve the same order as the parameters are listed below.